### PR TITLE
 Enable EnforceCodeStyleInBuild and fix findings

### LINKED ISSUE #2603: [Feature Request] Enforce Code Style In Build
**Describe the solution you'd like**

Use EnforceCodeStyleInBuild
https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild

Longer term, should align on code style with dotnet libraries, for example: https://github.com/dotnet/sdk/blob/main/.editorconfig

### PR DESCRIPTION
**Pull request by keegan-caruso**
# Enable EnforceCodeStyleInBuild and fix findings

Have to enable GenerateDocumentationFile for test csproj to work around
https://github.com/dotnet/roslyn/issues/41640

Fixes #2603

### CODE REVIEW

**Comment by armandRobled** [comment]
"Suscribirse"
MPSS.HA.1.1.c1-00015-1.484352.3.486950.2,MPSS.HA.1.1.c1-00015-1.484352.3.486950.2

**Comment by armandRobled** [comment]
🙏

Luis Armando reaccionó a través de Gmail
<https://www.google.com/gmail/about/?utm_source=gmail-in-product&utm_medium=et&utm_campaign=emojireactionemail#app>

**Review by pmaytak — APPROVED** [review]

**Review by iNinja — APPROVED** [review]
It appears we have a rule in our Wilson Github Action test that is failing due to a few files not having the usual header. From looking at them, it would appear that they may be exceptions to the rule.

**Comment by armandRobled** [comment]
Thanks for the tip!

**Comment by keegan-caruso** [comment]
Thanks @iNinja, I removed those files from the editorconfig rule for file headers.

File: .editorconfig [hunk 1]
```diff
@@ -169,6 +169,21 @@ dotnet_diagnostic.CA1303.severity = none
...
+# IDE0055: All formatting rules
+dotnet_diagnostic.IDE0055.severity = warning
+
+# CA1845: Use span-based 'string.Concat'
+dotnet_diagnostic.CA1845.severity = warning
+
+# CA1846: Prefer AsSpan over Substring
+dotnet_diagnostic.CA1846.severity = warning
+
+[{GlobalSuppressions.cs,TrimmingAttributes.cs}]
```
Comment (pmaytak) [inline .editorconfig:184]: I think we can still add the header to the GlobalSuppressions? Excluding TrimmingAttributes makes sense.

Comment (keegan-caruso) [reply .editorconfig:184]: GlobalSuppression is mostly a generated file

**Pull request merged**